yaks/lib/yaks/resource/form.rb

```diff
@@ -22,6 +22,10 @@ def fields_flat(fields = fields)
           end
         end
       end
+
+      def map_fields(fields = fields)
+        with(fields: fields.map { |field| yield(field) } )
+      end
```

Comment:
I was thinking map_fields would actually recurse into fieldsets so you don't have to care about nesting.